src/vehicle.cpp

```diff
@@ -195,10 +195,17 @@ bool Vehicle::NeedsServicing() const
 
 	/* Are we ready for the next service cycle? */
 	const Company *c = Company::Get(this->owner);
-	if (this->ServiceIntervalIsPercent() ?
-			(this->reliability >= this->GetEngine()->reliability * (100 - this->GetServiceInterval()) / 100) :
-			(this->date_of_last_service + this->GetServiceInterval() >= TimerGameEconomy::date)) {
-		return false;
+
+	/* There are three possible units for the service interval, which we handle individually. */
```

Comment:
This comment added no context at all. And is very prune to errors over time .. what if someone adds a forth? Will they remember to update this comment? :)